jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/IteratingCallback.java

```diff
@@ -198,6 +218,50 @@ protected void onCompleteFailure(Throwable cause)
     {
     }
 
+    /**
+     * Invoked when the overall task has been aborted.
+     * <p>
+     * Calls to this method are serialized with respect to {@link #process()}, {@link #onCompleteFailure(Throwable)}
+     * and {@link #onCompleted(Throwable)}.
+     * If this method is called, then {@link #onCompleteSuccess()} will never be called.
+     * <p>
+     * The default implementation of this method calls {@link #failed(Throwable)}.  Overridden implementations of
+     * this method SHOULD NOT call {@code super.onAborted(Throwable)}.
+     *
+     * @param cause The cause of the abort
+     */
+    protected void onAborted(Throwable cause)
+    {
+    }
+
+    /**
+     * Invoked when the overall task has completed.
+     * <p>
+     * Calls to this method are serialized with respect to {@link #process()} and {@link #onAborted(Throwable)}.
+     * The default implementation of this method will call either {@link #onCompleteSuccess()} or {@link #onCompleteFailure(Throwable)}
+     * thus implementations of this method should always call {@code super.onCompleted(Throwable)}.
+     *
+     * @param causeOrNull the cause of any {@link #abort(Throwable) abort} or {@link #failed(Throwable) failure},
+     * else {@code null} for {@link #succeeded() success}.
+     */
+    protected void onCompleted(Throwable causeOrNull)
```

Comment:
The fact that you can override this method as well as `onCompleteSuccess` and `onCompleteFailure`, plus the fact that you must no forget to call super is a bit confusing, especially since ordering isn't specified and depends on the call to super.

Reply:
I originally had ICB explicitly calling onCF or onCS directly.  @sbordet asked for this change to have just onC wired into ICB and have it call onCF or onCS.      I can see value both ways... so you two need to pick one.

I think the javadoc is clear on the ordering as it tells us that onC calls onCS or onCF.

Reply:
I'm fine with the smallest change at this point, so make this method private, and if necessary, let's open it up in the future.